### PR TITLE
testcase: Adds new unit test case for resource alicloud_vpc_vbr_ha alicloud_cen_instance alicloud_vpc_bgp_group

### PR DESCRIPTION
**Pull request by thisnihonglei**

### CODE REVIEW

File: alicloud/resource_alicloud_cen_instance_test.go [hunk 1]
```diff
@@ -660,3 +669,385 @@ func testAccCheckCenInstanceDestroy(s *terraform.State) error {
 
 	return nil
 }
+
+func TestAccAlicloudCenInstance_unit(t *testing.T) {
```
Comment (thisnihonglei) [inline alicloud/resource_alicloud_cen_instance_test.go:673]: 单测95.3%覆盖率

File: alicloud/resource_alicloud_vpc_bgp_group_test.go [hunk 2]
```diff
@@ -299,3 +309,324 @@ resource "alicloud_express_connect_virtual_border_router" "default" {
 }
 `, name)
 }
+
+func TestAccAlicloudVPCBgpGroup_unit(t *testing.T) {
```
Comment (thisnihonglei) [inline alicloud/resource_alicloud_vpc_bgp_group_test.go:313]: 单测97.8%覆盖率

File: alicloud/resource_alicloud_vpc_vbr_ha_test.go [hunk 3]
```diff
@@ -100,3 +109,239 @@ resource "alicloud_cen_instance_attachment" "default" {
 
 `, name, defaultRegionToTest)
 }
+
+func TestAccAlicloudVPCVbrHa_unit(t *testing.T) {
```
Comment (thisnihonglei) [inline alicloud/resource_alicloud_vpc_vbr_ha_test.go:113]: 单测96.4%覆盖率

File: alicloud/resource_alicloud_vpc_vbr_ha_test.go [hunk 4]
```diff
@@ -100,3 +109,239 @@ resource "alicloud_cen_instance_attachment" "default" {
...
+			}
+			return responseMock["DeleteNormal"]("")
+		})
+		patcheDescribeVpcVbrHa := gomonkey.ApplyMethod(reflect.TypeOf(&VpcService{}), "DescribeVpcVbrHa", func(*VpcService, string) (map[string]interface{}, error) {
+			return responseMock["NoRetryError"]("NoRetryError")
+		})
+		err := resourceAlicloudVpcVbrHaDelete(d, rawClient)
+		patches.Reset()
+		patcheDescribeVpcVbrHa.Reset()
+		assert.NotNil(t, err)
```
Comment (Wanghx0991) [inline alicloud/resource_alicloud_vpc_vbr_ha_test.go:295]: delete 这里应该是normal

File: alicloud/resource_alicloud_cen_instance_test.go [hunk 5]
```diff
@@ -660,3 +669,385 @@ func testAccCheckCenInstanceDestroy(s *terraform.State) error {
...
+			}
+			return responseMock["DeleteNormal"]("")
+		})
+		patcheDescribeCenInstance := gomonkey.ApplyMethod(reflect.TypeOf(&CbnService{}), "DescribeCenInstance", func(*CbnService, string) (map[string]interface{}, error) {
+			return responseMock["NoRetryError"]("NoRetryError")
+		})
+		err := resourceAlicloudCenInstanceDelete(d, rawClient)
+		patches.Reset()
+		patcheDescribeCenInstance.Reset()
+		assert.NotNil(t, err)
```
Comment (Wanghx0991) [inline alicloud/resource_alicloud_cen_instance_test.go:1001]: assert.Nil(t, err)

File: alicloud/resource_alicloud_vpc_vbr_ha_test.go [hunk 6]
```diff
@@ -100,3 +109,239 @@ resource "alicloud_cen_instance_attachment" "default" {
...
+		err := resourceAlicloudVpcVbrHaCreate(dCreate, rawClient)
+		patches.Reset()
+		assert.Nil(t, err)
+	})
+
+	// Set ID for Update and Delete Method
+	d.SetId("MockVbrHaId")
+	// Update
+	t.Run("UpdateClientAbnormal", func(t *testing.T) {
+		patches := gomonkey.ApplyMethod(reflect.TypeOf(&connectivity.AliyunClient{}), "NewVpcClient", func(_ *connectivity.AliyunClient) (*client.Client, error) {
```
[the remark this answers is not in the document]
Comment (thisnihonglei) [reply alicloud/resource_alicloud_vpc_vbr_ha_test.go]: done

File: alicloud/resource_alicloud_vpc_vbr_ha_test.go (hunk 4 above)
Comment (thisnihonglei) [reply alicloud/resource_alicloud_vpc_vbr_ha_test.go:286]: done

File: alicloud/resource_alicloud_vpc_bgp_group_test.go [hunk 7]
```diff
@@ -299,3 +309,324 @@ resource "alicloud_express_connect_virtual_border_router" "default" {
...
+				return responseMock["NoRetryError"]("NonRetryableError")
+			}
+			return responseMock["DeleteNormal"]("")
+		})
+		patcheDescribeVpcVbrHa := gomonkey.ApplyMethod(reflect.TypeOf(&VpcService{}), "DescribeVpcBgpGroup", func(*VpcService, string) (map[string]interface{}, error) {
+			return responseMock["NoRetryError"]("NoRetryError")
+		})
+		err := resourceAlicloudVpcBgpGroupDelete(d, rawClient)
+		patches.Reset()
+		patcheDescribeVpcVbrHa.Reset()
```
[the remark this answers is not in the document]
Comment (thisnihonglei) [reply alicloud/resource_alicloud_vpc_bgp_group_test.go:579]: done

File: alicloud/resource_alicloud_cen_instance_test.go (hunk 5 above)
Comment (thisnihonglei) [reply alicloud/resource_alicloud_cen_instance_test.go:1001]: done

**Pull request merged**